### PR TITLE
Add suport to the Mroonga engine

### PR DESCRIPTION
**Pull request by williamdes**
Add suport to the Mroonga engine

See: #16945

### CODE REVIEW

**Comment by williamdes** [comment]
cc @josepsanzcamp

File: test/classes/StorageEngineTest.php [hunk 1]
```diff
@@ -319,4 +322,111 @@ public function testResolveTypeSize(): void
...
+                (string) json_encode([
+                    'mroonga_operations' => [],
+                    'mroonga_operations.type' => [],
+                    'mroonga_operations.table' => [],
+                    'mroonga_operations.record' => [],
+                    'idx_correo' => [],
+                    //'idx_correo.id' => [],
+                    //'idx_correo.search' => [],
+                    'idx_correo#idx_correo_search' => [],
+                    //'idx_correo#idx_correo_search.index' => [],
```
Comment (williamdes) [inline test/classes/StorageEngineTest.php:364]: @josepsanzcamp it seems like the object list command outputs a long list, could we filter on the `type` ?
See the test data below

Comment (josepsanzcamp) [reply test/classes/StorageEngineTest.php:364]: Hi @williamdes

It's a good idea your optimization.

I have applied it in the last commit.

Thanks by your suggestion.

Josep.

Comment (williamdes) [reply test/classes/StorageEngineTest.php:364]: Hi @josepsanzcamp 
Can you push it or paste it here as a git diff ?

Comment (josepsanzcamp) [reply test/classes/StorageEngineTest.php:364]: Hi @williamdes

Sorry, but I have detected that I did not pushed the last commit!

See the commit 4af2f9e6c5602cf81e23ec426d1d59693d473b59 for more information about the optimization.

Sorry for the mistake!!!

Josep.

Comment (williamdes) [reply test/classes/StorageEngineTest.php]: Thank you !
I could push back and re-open #16945
This one will auto close when I merge your work

**Pull request merged**